tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+    """S3 copy object class."""
+    # pylint: disable=too-many-arguments
+    # pylint: disable-msg=too-many-statements
+    # pylint: disable=too-many-arguments
+    # pylint: disable-msg=too-many-locals
+    @classmethod
+    def setup_class(cls):
+        """
+        Setup_class will be invoked prior to each test case.
+
+        It will perform all prerequisite test suite steps if any.
+        """
+        cls.log = logging.getLogger(__name__)
+        cls.log.info("STARTED: Class setup.")
+        cls.csm_obj = csm_api_factory("rest")
+        cls.ha_obj = HAK8s()
+        cls.config = CSMConfigsCheck()
+        cls.log.info("Initiating Rest Client ...")
+        cls.csm_conf = configmanager.get_config_wrapper(fpath="config/csm/test_rest_capacity.yaml")
+        cls.rest_resp_conf = configmanager.get_config_wrapper(
+            fpath="config/csm/rest_response_data.yaml")
+        cls.created_iam_users = set()
+        cls.buckets_created = []
+        cls.akey = None
+        cls.skey = None
+        cls.src_bkt = None
+        cls.dest_bkt1 = None
+        cls.obj = None
+        cls.user_id = None
+        setup_ready = cls.config.check_predefined_csm_user_present()
+        if not setup_ready:
+            setup_ready = cls.config.setup_csm_users()
+        assert setup_ready
+        resp = cls.ha_obj.get_config_value(cls.csm_obj.master)
+        if resp[0]:
+            cls.nvalue = int(resp[1]['cluster']['storage_set'][0]['durability']['sns']['data'])
+        cls.aligned_size = 4 * cls.nvalue
+        cls.test_file = "mp_obj"
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "test_copy_object")
+        cls.mp_obj_path = os.path.join(cls.test_dir_path, cls.test_file)
+        if not path_exists(cls.test_dir_path):
+            make_dirs(cls.test_dir_path)
+            cls.log.info("Created path: %s", cls.test_dir_path)
+        cls.log.info("ENDED: Class setup.")
+
+    def setup_method(self):
+        """
+        Setup_method will be invoked before running each test case.
+
+        It will perform all prerequisite steps required for test execution.
+        It will create a bucket and upload an object to that bucket.
+        """
+        self.log.info("STARTED: Test setup.")
+        self.log.info("Step 1: Create a user (user1)")
+        payload = self.csm_obj.iam_user_payload_rgw("random")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        self.user_id = resp.json()["keys"][0]['user']
+        self.created_iam_users.add(self.user_id)
+        resp1 = self.csm_obj.compare_iam_payload_response(resp, payload)
+        self.log.info("Printing response %s", resp1)
+        assert resp1[0], resp1[1]
+        self.akey = resp.json()["keys"][0]["access_key"]
+        self.skey = resp.json()["keys"][0]["secret_key"]
+        self.src_bkt = "src1"
```

Comment:
Same as line 83, let's add a timestamp to file/resource names. Please refactor this wherever it applies in this module.

Reply:
I feel it is not necessary for bucket and oject name as these names are getting initialised anc cleaned after every function execution. Therefore, it wont create issues of duplicate name exists.
Also keeping simpler names for copy-object's bucket is helpful to debug from names like src1/obj1 to src1/obj2.

Reply:
Logging should help keep track of bucket/object names per test, so that shouldn't be an issue. We cannot always assume that cleanup/teardown will succeed.

Reply:
addressed

Reply:
Let's update bucket name with timestamp as well, as per our offline discussion

Reply:
done